client/my-sites/themes/themes-selection.jsx

```diff
@@ -34,6 +31,35 @@ const ThemesSelection = React.createClass( {
 		tier: React.PropTypes.string,
 	},
 
+	getDefaultProps() {
+		return { search: '' };
+	},
+
+	doSearch( searchString ) {
+		const filterRegex = /\w+\:\s*([\w-]+)/g;
```

Comment:
Although I don't believe there are currently any numeric chars in the taxonomies, I don't think there is anything to prevent that in the future, so seems good to go with `\w`.